src/send-commercial-metrics.spec.ts

```diff
@@ -36,97 +43,224 @@ const defaultMetrics = {
 		value,
 		writable: true,
 	});
 };
 
-describe('send commercial metrics code', () => {
-	const sendBeacon = jest.fn().mockReturnValue(true);
+beforeEach(() => {
+	reset();
+	jest.resetAllMocks();
+});
+
+describe('send commercial metrics', () => {
 	Object.defineProperty(navigator, 'sendBeacon', {
 		configurable: true,
 		enumerable: true,
-		value: sendBeacon,
+		value: jest.fn(),
 		writable: true,
 	});
 
 	const mockConsoleWarn = jest
 		.spyOn(console, 'warn')
 		.mockImplementation(() => false);
 
-	it('can send commercial metrics when the page is hidden', () => {
-		const metricsSent = mockSendMetrics();
-		setVisibility();
+	it('sends metrics when the page is hidden, user is in sampling group, and consent is given', async () => {
+		mockOnConsent(tcfv2AllConsent);
+
+		await initCommercialMetrics({
+			pageViewId: PAGE_VIEW_ID,
+			browserId: BROWSER_ID,
+			isDev: IS_NOT_DEV,
+			adBlockerInUse: ADBLOCK_NOT_IN_USE,
+			sampling: USER_IN_SAMPLING,
+		});
+
+		setVisibility('hidden');
 		global.dispatchEvent(new Event('visibilitychange'));
 
-		expect(metricsSent).toEqual(true);
 		expect((navigator.sendBeacon as jest.Mock).mock.calls).toEqual([
 			[Endpoints.PROD, JSON.stringify(defaultMetrics)],
 		]);
 	});
 
-	it('commercial metrics not sent when window is visible', () => {
-		const metricsSent = mockSendMetrics();
+	it('does not send metrics when page is visible', async () => {
+		mockOnConsent(tcfv2AllConsent);
+
+		await initCommercialMetrics({
+			pageViewId: PAGE_VIEW_ID,
+			browserId: BROWSER_ID,
+			isDev: IS_NOT_DEV,
+			adBlockerInUse: ADBLOCK_NOT_IN_USE,
+			sampling: USER_IN_SAMPLING,
+		});
+
 		setVisibility('visible');
 		global.dispatchEvent(new Event('visibilitychange'));
 
-		expect(metricsSent).toEqual(false);
 		expect((navigator.sendBeacon as jest.Mock).mock.calls).toEqual([]);
 	});
 
+	it('does not send metrics when user is not in sampling group', async () => {
+		mockOnConsent(tcfv2AllConsent);
+
+		await initCommercialMetrics({
+			pageViewId: PAGE_VIEW_ID,
+			browserId: BROWSER_ID,
+			isDev: IS_NOT_DEV,
+			adBlockerInUse: ADBLOCK_NOT_IN_USE,
+			sampling: USER_NOT_IN_SAMPLING,
+		});
+
+		setVisibility('hidden');
+		global.dispatchEvent(new Event('visibilitychange'));
+
+		expect((navigator.sendBeacon as jest.Mock).mock.calls).toEqual([]);
+	});
+
+	it('does not send metrics when consent does not include purpose 8', async () => {
+		mockOnConsent(tcfv2AllConsentExceptPurpose8);
+
+		await initCommercialMetrics({
+			pageViewId: PAGE_VIEW_ID,
+			browserId: BROWSER_ID,
+			isDev: IS_NOT_DEV,
+			adBlockerInUse: ADBLOCK_NOT_IN_USE,
+			sampling: USER_IN_SAMPLING,
+		});
+
+		setVisibility('hidden');
+		global.dispatchEvent(new Event('visibilitychange'));
+
+		expect((navigator.sendBeacon as jest.Mock).mock.calls).toEqual([]);
+	});
+
+	it('sends metrics when consent is non-TCFv2 (i.e. when user is in USA or Australia)', async () => {
+		mockOnConsent(nonTcfv2Consent);
```

Comment:
Testing that metrics are sent for CCPA and Aus users regardless of consent state, it might be more complete to add another test case where `doNotSell` is true.

Reply:
Yup good shout, I will add!